pkg/kubelet/kuberuntime/helpers.go

```diff
@@ -0,0 +1,169 @@
+func buildContainerName(podName, podNamespace, podUID string, container *api.Container) (string, string, string) {
+	containerName := container.Name + "." + strconv.FormatUint(kubecontainer.HashContainer(container), 16)
+	stableName := fmt.Sprintf("%s_%s_%s_%s_%s",
+		kubePrefix,
+		containerName,
+		podName,
+		podNamespace,
+		podUID,
+	)
+	UID := fmt.Sprintf("%08x", rand.Uint32())
+	return stableName, fmt.Sprintf("%s_%s", stableName, UID), UID
+}
+
+// parseContainerName unpacks a container name, returning the pod name, namespace, UID and container name
+func parseContainerName(name string) (podName, podNamespace, podUID, containerName string, hash uint64, err error) {
+	parts := strings.Split(name, "_")
+	if len(parts) == 0 || parts[0] != kubePrefix {
+		err = fmt.Errorf("failed to parse container name %q into parts", name)
+		return "", "", "", "", 0, err
+	}
+	if len(parts) < 6 {
+		glog.Warningf("Found a container with the %q prefix, but too few fields (%d): %q", kubePrefix, len(parts), name)
+		err = fmt.Errorf("container name %q has fewer parts than expected %v", name, parts)
+		return "", "", "", "", 0, err
+	}
+
+	nameParts := strings.Split(parts[1], ".")
+	containerName = nameParts[0]
+	if len(nameParts) > 1 {
+		hash, err = strconv.ParseUint(nameParts[1], 16, 32)
+		if err != nil {
+			glog.Warningf("invalid container hash %q in container %q", nameParts[1], name)
+		}
+	}
+
+	return parts[2], parts[3], parts[4], containerName, hash, nil
+}
+
+// toRuntimeProtocol converts api.Protocol to runtimeApi.Protocol
+func toRuntimeProtocol(protocol api.Protocol) runtimeApi.Protocol {
+	switch protocol {
+	case api.ProtocolTCP:
+		return runtimeApi.Protocol_TCP
+	case api.ProtocolUDP:
+		return runtimeApi.Protocol_UDP
+	}
+
+	glog.Warningf("Unknown protocol %q: defaulting to TCP", protocol)
+	return runtimeApi.Protocol_TCP
+}
+
+// milliCPUToShares converts milliCPU to CPU shares
+func milliCPUToShares(milliCPU int64) int64 {
+	if milliCPU == 0 {
+		// Docker converts zero milliCPU to unset, which maps to kernel default
+		// for unset: 1024. Return 2 here to really match kernel default for
+		// zero milliCPU.
+		return minShares
+	}
+	// Conceptually (milliCPU / milliCPUToCPU) * sharesPerCPU, but factored to improve rounding.
+	shares := (milliCPU * sharesPerCPU) / milliCPUToCPU
+	if shares < minShares {
+		return minShares
+	}
+	return shares
+}
+
+// milliCPUToQuota converts milliCPU to CFS quota and period values
+func milliCPUToQuota(milliCPU int64) (quota int64, period int64) {
+	// CFS quota is measured in two values:
+	//  - cfs_period_us=100ms (the amount of time to measure usage across)
+	//  - cfs_quota=20ms (the amount of cpu time allowed to be used across a period)
+	// so in the above example, you are limited to 20% of a single CPU
+	// for multi-cpu environments, you just scale equivalent amounts
+	if milliCPU == 0 {
+		// take the default behavior from docker
```

Comment:
Remove this comment since it's docker specific.

@dchen1107 @vishh should we set a default so that it's consistent across runtimes?